assets/src/design-system/components/dropDown/index.js

```diff
@@ -69,6 +86,31 @@ export const DropDown = ({
     [isOpen]
   );
 
+  const handleDismissMenu = useCallback(() => {
+    isOpen.set(false);
+    selectRef.current.focus();
+  }, [isOpen]);
+
+  const handleMenuItemClick = useCallback(
+    (event, menuItem) => {
+      onMenuItemClick && onMenuItemClick(event, menuItem);
+
+      if (!isKeepMenuOpenOnSelection) {
+        handleDismissMenu();
+      }
+    },
+    [handleDismissMenu, isKeepMenuOpenOnSelection, onMenuItemClick]
+  );
+
+  useEffect(() => {
+    if (selectRef?.current) {
+      const { height = null } = selectRef?.current?.getBoundingClientRect();
+      setAnchorHeight(height);
+    }
+  }, []);
+
+  const listId = `list-${uuidv4()}`;
```

Comment:
does this need to be memoized at all so it's not producing a new id every render?